### ISSUE TITLE
Document operating system dependencies

### ISSUE THREAD

**Issue by stain** [eitanf/sysconf#21]
It seems the installation is meant to be done in Linux, but this is not documented in the README.

In addition, even when using a fresh Ubuntu 20.04, there seem to be extra system dependencies that needs to be pre-installed that are not documented by the README.

For instance, I get an error for installing `kableExtra` but the message is lost in the verbose output of `make prereq`, this causes an error on `make`:

```
Quitting from lines 33-116 (_main.Rmd)
Error in library("kableExtra") : there is no package called 'kableExtra'
Calls: <Anonymous> ... withCallingHandlers -> withVisible -> eval -> eval -> library
```

I believe this is caused by `freetype` libraries missing when `kableExtra` install is attempted:

```
Downloading package from url: http://cran.r-project.org/src/contrib/Archive/kableExtra/kableExtra_1.3.2.tar.gz
systemfonts (NA -> 1.0.2) [CRAN]
svglite     (NA -> 2.0.0) [CRAN]
Installing 2 packages: systemfonts, svglite
trying URL 'http://cran.r-project.org/src/contrib/systemfonts_1.0.2.tar.gz'
Content type 'application/x-gzip' length 80693 bytes (78 KB)
==================================================
downloaded 78 KB

trying URL 'http://cran.r-project.org/src/contrib/svglite_2.0.0.tar.gz'
Content type 'application/x-gzip' length 122260 bytes (119 KB)
==================================================
downloaded 119 KB

begin installing package ‘systemfonts’
make: *** [Makefile:4: systemfonts.ts] Error 1
make: Target 'all' not remade because of errors.
* installing *source* package ‘systemfonts’ ...
** package ‘systemfonts’ successfully unpacked and MD5 sums checked
** using staged installation
Package fontconfig was not found in the pkg-config search path.
Perhaps you should add the directory containing `fontconfig.pc'
to the PKG_CONFIG_PATH environment variable
No package 'fontconfig' found
Package freetype2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `freetype2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'freetype2' found
Using PKG_CFLAGS=
Using PKG_LIBS=-lfontconfig -lfreetype
** libs
make[1]: Entering directory '/tmp/Rtmp3wX408/R.INSTALLa2d560d268f/systemfonts/src'
rm -f systemfonts.so caches.o cpp11.o dev_metrics.o font_matching.o font_registry.o ft_cache.o string_shape.o font_metrics.o font_fallback.o string_metrics.o emoji.o cache_store.o init.o unix/FontManagerLinux.o
x86_64-conda_cos6-linux-gnu-c++ -std=gnu++11 -I"/home/stain/miniconda3/envs/test/lib/R/include" -DNDEBUG  -I"/home/stain/miniconda3/envs/test/lib/R/library/cpp11/include" -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/stain/miniconda3/envs/test/include -I/home/stain/miniconda3/envs/test/include -Wl,-rpath-link,/home/stain/miniconda3/envs/test/lib  -fpic  -fvisibility-inlines-hidden  -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/stain/miniconda3/envs/test/include -fdebug-prefix-map=/tmp/build/80754af9/r-base_1589917437985/work=/usr/local/src/conda/r-base-3.6.1 -fdebug-prefix-map=/home/stain/miniconda3/envs/test=/usr/local/src/conda-prefix  -c caches.cpp -o caches.o
In file included from caches.h:7:0,
                 from caches.cpp:1:
ft_cache.h:9:10: fatal error: ft2build.h: No such file or directory
 #include <ft2build.h>
          ^~~~~~~~~~~~
compilation terminated.
make[1]: *** [/home/stain/miniconda3/envs/test/lib/R/etc/Makeconf:175: caches.o] Error 1
make[1]: Leaving directory '/tmp/Rtmp3wX408/R.INSTALLa2d560d268f/systemfonts/src'
ERROR: compilation failed for package ‘systemfonts’
* removing ‘/home/stain/miniconda3/envs/test/lib/R/library/systemfonts’
cat: svglite.out: No such file or directory
```

Could you add installation requirements from a blank Ubuntu 20.04 or similar, including which R system packages to install? See also #20 on `pandoc` and `gtar`

**Comment by stain** [comment]
Also on python dependencies:

```
python3 /home/stain/src/sysconf/src/gather_papers.py CIDR PPoPP HPCA NDSS FAST NSDI ASPLOS VEE ICPE ISPASS EuroSys HotOS CCGrid SIGMOD PODS SYSTOR HCW IPDPS SIGMETRICS ISC ISCA CLOUD HPDC HotCloud HotStorage ATC SPAA PODC ICPP SIGCOMM EuroPar Cluster PACT MASCOTS SOCC IISWC SP MICRO MobiCom IGSC ICAC CCS SOSP IMC SC Middleware CoNEXT HiPC HPCC ICDM KDD SIGIR  OOPSLA PLDI SLE HotI
Traceback (most recent call last):
  File "/home/stain/src/sysconf/src/authorship_graph.py", line 11, in <module>
    import tidydata
  File "/home/stain/src/sysconf/src/tidydata.py", line 19, in <module>
    from shared_utils import round_values
  File "/home/stain/src/sysconf/src/shared_utils.py", line 7, in <module>
    from dateutil.relativedelta import relativedelta
ModuleNotFoundError: No module named 'dateutil'
Traceback (most recent call last):
  File "/home/stain/src/sysconf/src/gather_papers.py", line 9, in <module>
    import tidydata
  File "/home/stain/src/sysconf/src/tidydata.py", line 19, in <module>
    from shared_utils import round_values
  File "/home/stain/src/sysconf/src/shared_utils.py", line 7, in <module>
    from dateutil.relativedelta import relativedelta
ModuleNotFoundError: No module named 'dateutil'
```

**Comment by eitanf** [comment]
Thank you for your suggestions. The installation instructions were indeed obsolete and incomplete.
I've updated README.md and added INSTALL.md to rectify this problem, having installed everything on a fresh Ubuntu VM. The instructions should allow anyone starting from scratch on either Ubuntu 20.04 or MacOS 15 to get to a point where they can compile any paper and compute any features file.
Although I did not encounter the specific issue you did with kableExtra, I did add the suggestion for `tar` to the INSTALL.md file.

**Comment by stain** [comment]
Thanks for that, very helpful. Just tested in Ubuntu 20.04 container on WSL2, and the install now included `kableExtra` (probably as system deps now were present).  I also don't seem to need the `TAR` workaround - that may be specific to the conda distribution of R I used first time.

I got just one warning:

```
Warning messages:
1: package ‘MUMIn’ is not available (for R version 3.6.3)
2: Perhaps you meant ‘MuMIn’ ?
```

I installed that manually for now:

```
> install.packages("MuMIn")
Installing package into ‘/home/stain/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/MuMIn_1.43.17.tar.gz'
```

Next error:

```
/usr/bin/pandoc +RTS -K512m -RTS _main.knit.md --to latex --from markdown+tex_math_single_backslash-autolink_bare_uris --output _main.tex --lua-filter /home/stain/R/x86_64-pc-linux-gnu-library/3.6/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /home/stain/R/x86_64-pc-linux-gnu-library/3.6/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /home/stain/R/x86_64-pc-linux-gnu-library/3.6/rmarkdown/rmarkdown/lua/latex-div.lua --metadata-file /tmp/RtmpTcN5Hf/file2b31564f955b --self-contained --table-of-contents --toc-depth 2 --template /home/stain/R/x86_64-pc-linux-gnu-library/3.6/rticles/rmarkdown/templates/plos/resources/template.tex --number-sections --highlight-style tango --pdf-engine pdflatex --wrap preserve --include-in-header /tmp/RtmpTcN5Hf/rmarkdown-str2b3170c63ed5.html --variable tables=yes --standalone --filter /usr/bin/pandoc-citeproc
pandoc-citeproc: PandocResourceNotFound "plos.csl"
Error running filter /usr/bin/pandoc-citeproc:
Filter returned error status 1
Error: pandoc document conversion failed with error 83
In addition: There were 16 warnings (use warnings() to see them)
Please delete _main.Rmd after you finish debugging the error.
Execution halted
make[1]: *** [Makefile:14: gender-gap.pdf] Error 1
make[1]: Leaving directory '/home/stain/src/sysconf/pubs/gender-gap'
```
Fixed by `cp prestige/plos.csl gender-gap/`

Now `prestige` fails:

```
Error in library("corrplot") : there is no package called 'corrplot'
Calls: <Anonymous> ... withCallingHandlers -> withVisible -> eval -> eval -> library

Please delete _main.Rmd after you finish debugging the error.
Execution halted
make[1]: *** [Makefile:14: prestige.pdf] Error 1
make[1]: Leaving directory '/home/stain/src/sysconf/pubs/prestige'
```

`corrplot` is listed in `./prestige/deps.R`  but not used by your new install commands. I ran `Rscript prestige/deps.R` manually as workaround - should this be sourced from dependencies.R?

So remaining:

- [ ] add `corrplot` to `dependencies.R`
- [ ] add rest of `prestige/deps.R` to `dependencies.R`?
- [ ] `MuMIn` typo ~MUMIn~
- [ ] Copy/link `gender-gap/plos.csl`

After this they all built nicely! (except `diversity-survey` due to confidential data - is it possible to try dummy-data as placeholder? See pull request #22 .

**Comment by eitanf** [comment]
Thanks again, this is helpful.
I did already fix the three issues you pointed out in yesterday's push. Are you not seeing it in pubs/dependencies.R?  But I didn't add the rest of prestige/deps.R (I think on RColorBrewer is missing)? So by all means, do add those instead in our PR.
As for `diversity-survey` I think it's better to make it abundantly clear that the analysis cannot be reproduced rather then mislead people into a PDF with wrong/dummy data. It's not reproducible either way, but the code is still all there to assess the validity of the analysis.
